engine/battle/core.asm

```diff
@@ -7139,10 +7139,20 @@ GiveExperiencePoints:
 	ld [hl], a
 	jr nc, .no_exp_overflow
 	dec hl
+	push bc
+	push af
+	ld a, [hl]
+	and EXP_MASK
+	ld b, a
 	inc [hl]
```

Comment:
Theoretically if the value at `[hl]` were `%11_111111` -- i.e. the mon has over 4 million exp -- then incrementing `[hl]` could affect the high bits of the caught time. If you're assuming exp will never get that high it's fine.

Reply:
For now i'm assuming that mon's will remain below 4 million experience.